internal/config/validator.go

```diff
@@ -58,13 +56,51 @@ func (v *validator) Validate(resources ...client.ObjectList) error {
 			clusterResources.Nodes = append(clusterResources.Nodes, list.Items...)
 		}
 	}
+	clusterResources = resetTransientErrorsFields(clusterResources)
 	_, err := For(clusterResources, v.validate)
-	if errors.As(err, &TransientError{}) { // we do not want to make assumption on ordering in webhooks.
-		return nil
-	}
 	return err
 }
 
+// resetting the fields that lead to Transienterror
+func resetTransientErrorsFields(clusterResources ClusterResources) ClusterResources {
+	for i := range clusterResources.Peers {
+		clusterResources.Peers[i].Spec.BFDProfile = ""
+		clusterResources.Peers[i].Spec.PasswordSecret.Name = ""
+	}
+	for i, bgpAdv := range clusterResources.BGPAdvs {
+		var communities []string
+		for _, community := range bgpAdv.Spec.Communities {
+			if existCommunity(community, clusterResources.Communities) {
```

Comment:
wouldn't this make us miss real failures like a community with an invalid value?
like "234.4"

Reply:
In my opinion, no, because the verification of whether the values are correct or not should occur in the preceding functions. In the functions here, I only delete or change to an empty string. 
@liornoy

Reply:
After discussing with Lior, I agree.
Thanks @liornoy  for your attention here.